store.go

```diff
@@ -698,9 +702,20 @@ func (s *store) load() error {
 		return err
 	}
 	s.containerStore = rcs
+
+	s.lockRoot = filepath.Join(s.runRoot, driverPrefix+"locks")
+	if err := os.MkdirAll(s.lockRoot, 0700); err != nil {
+		return err
+	}
+
 	return nil
 }
 
+// GetDigestLock returns a digest-specific Locker.
+func (s *store) GetDigestLock(d digest.Digest) (Locker, error) {
+	return GetLockfile(filepath.Join(s.lockRoot, d.Encoded()))
```

Comment:
Is this intentionally dropping the algorithm? (In principle, this may cause a few temporary collisions, which does not really hurt. OTOH adding the algorithm and having a bit longer file names does not hurt either.)

Reply:
I did not consider the potential of collisions, great catch!